mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+        std::cout
+        << " \n"
+        << " version: v0.1.0\n" 
+        << std::endl;
+
+        
+    }
+
+    // The input file 
+    if (input_fileGiven) {
+
+        // At the moment, this will just be a print statement
+        std::cout << "The input file to the program is " << input_file << " ." << std::endl;
+        
+    }
+    else{
+        std::cout << "The input file must be specified. Use -h / --help for more information." << std::endl;
+         
+
+    }
+
+    // The output file
+    if (output_fileGiven) {
+
+        // At the moment, this will just be a print statement
+        std::cout << "The output file to the program is " << output_file << " ." << std::endl;
+        
+    }
+    else{
+
+        std::cout << "The output file must be specified. Use -h / --help for more information." << std::endl;
+         
+
+    }
+
+    // I'm keeping the user input part of my code for now.
+
+    // Initialising variables
+    char in_char{'x'};
+    std::string out_str{""};
+
+    // Take each letter from user input and in each case:
+    while(std::cin >> in_char)
+    {
+            
+        //- Convert to upper case
+        if(std::isalpha(in_char)){
+            out_str += std::toupper(in_char);
+            continue;
+        }
+
+        //- Change numbers to words
+        switch(in_char){
+
+            case '0':
+                out_str += "ZERO";
+                break;
+            
+            case '1':
+                out_str += "ONE";
+                break;
+            
+            case '2':
+                out_str += "TWO";
+                break;
+            
+            case '3':
+                out_str += "THREE";
+                break;
+
+            case '4':
+                out_str += "FOUR";
+                break;
+
+            case '5':
+                out_str += "FIVE";
+                break;
+
+            case '6':
+                out_str += "SIX";
+                break;
+
+            case '7':
+                out_str += "SEVEN";
+                break;
+
+            case '8':
+                out_str += "EIGHT";
+                break;
+
+            case '9':
+                out_str += "NINE";
+                break;
+            
+            default:
+                // Do nothing
+                break;
+        }
+
+        //- Ignore any other (non-alpha) characters
+        
+    }
+
+
+
+    // print out the new string
+    std::cout << out_str << std::endl;
+    
+}
+
+// I'm happy with submitting this as my attempt at the homework. 
+// The only problem that i know that still exists with my code is that, for instance, if typed
+// ./mpags-cipher -i -o my_output_file 
+// I know that my code would list -o as the input file and this is incorrect. 
+// I'm not sure how to get it to recognise that -o isn't a file.
+// I'm not sure i can do that at this stage because I'm not checking anything about the form of the argument that comes after -i or -o.
```

Comment:
Very good that you're thinking about these nuances :) Indeed, there is a limit to the error conditions that you can reasonably check for.  In fact, is it reasonable to disallow `-o` as the name of then input file?  The Linux operating system will allow such a filename.  Granted, it is a very strange choice but it's probably not up to us as the programmer to police the names of files that users might want to use.  So I think your check that there is a following argument is good enough (modulo the comment above about incrementing the loop counter).